cli/cmd/envoy-stats/envoy_stats.go

```diff
@@ -0,0 +1,191 @@
+package envoy_stats
+
+import (
+	"errors"
+	"fmt"
+	"github.com/hashicorp/consul-k8s/cli/common"
+	"github.com/hashicorp/consul-k8s/cli/common/flag"
+	"github.com/hashicorp/consul-k8s/cli/common/terminal"
+	"github.com/hashicorp/consul-k8s/cli/helm"
+	helmCLI "helm.sh/helm/v3/pkg/cli"
+	"io"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"net/http"
+	"strconv"
+	"sync"
+)
+
+const envoyAdminPort = 19000
+
+type Command struct {
+	*common.BaseCommand
+
+	helmActionsRunner helm.HelmActionsRunner
+
+	kubernetes kubernetes.Interface
+
+	restConfig *rest.Config
+
+	set *flag.Sets
+
+	flagKubeConfig  string
+	flagKubeContext string
+	flagNamespace   string
+	flagPod         string
+
+	once sync.Once
+	help string
+}
+
+func (c *Command) init() {
+	c.set = flag.NewSets()
+
+	f := c.set.NewSet("Global Options")
+	f.StringVar(&flag.StringVar{
+		Name:    "kubeconfig",
+		Aliases: []string{"c"},
+		Target:  &c.flagKubeConfig,
+		Default: "",
+		Usage:   "Path to kubeconfig file.",
+	})
+	f.StringVar(&flag.StringVar{
+		Name:    "context",
+		Target:  &c.flagKubeContext,
+		Default: "",
+		Usage:   "Kubernetes context to use.",
+	})
+	f.StringVar(&flag.StringVar{
+		Name:    "namespace",
+		Target:  &c.flagNamespace,
+		Default: "",
+		Usage:   "Namespace of pod",
+	})
+	f.StringVar(&flag.StringVar{
+		Name:    "pod",
+		Target:  &c.flagPod,
+		Default: "",
+		Usage:   "Pod for which envoy stats need to be captured",
+	})
+
+	c.help = c.set.Help()
+}
+
+// validateFlags checks the command line flags and values for errors.
+func (c *Command) validateFlags() error {
+	if len(c.set.Args()) > 0 {
+		return errors.New("should have no non-flag arguments")
+	}
+	return nil
+}
+
+func (c *Command) Run(args []string) int {
+	c.once.Do(c.init)
+
+	if err := c.set.Parse(args); err != nil {
+		c.UI.Output(err.Error())
+		return 1
+	}
+
+	if err := c.validateFlags(); err != nil {
+		c.UI.Output(err.Error())
+		return 1
+	}
+
+	if c.flagPod == "" || c.flagNamespace == "" {
+		c.UI.Output("namespace and pod name are required")
+		return 1
+	}
```

Comment:
It's more common in the Kubernetes ecosystem to use the namespace provided in the Kubeconfig if none is explicityly provided. This is how `kubectl` does it and we followed that model with all of the existing commands. If a user provides a value for `namespace`, it should be used. If not, fall back to whatever the value is in the Kubeconfig.

You can follow the pattern here: https://github.com/hashicorp/consul-k8s/blob/e50c8e20f039f33e0261409da91870daafc28aaa/cli/cmd/proxy/read/command.go#L298